ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/Sink.java

```diff
@@ -53,30 +55,35 @@ public static void main(String[] args) {
           Env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
           Env.getInt(BATCH_MAX_MESSAGES, 1_000_000), // default 1M messages
           Env.getDuration(BATCH_MAX_DELAY, "10m"), // default 10 minutes
-          gcsPrefix, format);
+          gcsPrefix, format)::apply;
```

Comment:
I don't understand why we're now calling `apply` here.

Reply:
because `Function<PubsubMessage, CompletableFuture<?>>` is incompatible with `Function<PubsubMessage, CompletableFuture<Void>>` and `Function<PubsubMessage, CompletableFuture<String>>`, due to the nesting levels, but I just realized that I can wrap `Pubsub.Write` so that it is `Function<PubsubMessage, CompletableFuture<Void>>`.